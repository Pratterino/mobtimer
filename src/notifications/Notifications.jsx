import React, { useEffect, useReducer } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Emojione } from 'react-emoji-render';
import { askForNotificationPermission, hasAcceptedNotifications } from './../NotificationManager';
import './Notifications.scss';

function Notifications() {
    const [, forceUpdate] = useReducer(x => x + 1, 0);

    useEffect(() => {
        askForNotificationPermission(() => forceUpdate());
    }, []);

    return (
        <div className="notifications">
            {!hasAcceptedNotifications() && (
                <div className="notifications__permissions">
                    <Emojione text=":information_source: :point_up: Please accept Notifications to get notified when the next user is due!" />
                </div>
            )}
        </div>
    );
}

const mapStateToProps = state => ({
    notifications: state.notifications,
});
const mapDispatchToProps = dispatch => bindActionCreators({}, dispatch);

export const unwrapped = Notifications;
export default connect(mapStateToProps, mapDispatchToProps)(Notifications);
