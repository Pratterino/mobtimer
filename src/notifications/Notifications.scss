.notifications {
  background: var(--background);

  .notifications__permissions {
    border: 1px solid var(--highlight-color);
    color: red;
  }
}
