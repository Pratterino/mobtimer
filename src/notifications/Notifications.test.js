import React from 'react';
import { unwrapped as UnwrappedNotifications } from './Notifications';
import { shallow } from 'enzyme';

describe('Notifications', () => {
    let wrapper;

    const renderComponent = props => (wrapper = shallow(<UnwrappedNotifications {...props} />));

    beforeEach(() => {
        renderComponent();
    });

    describe('render', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
