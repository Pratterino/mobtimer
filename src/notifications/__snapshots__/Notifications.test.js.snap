// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications render should match snapshot 1`] = `
<div
  className="notifications"
>
  <div
    className="notifications__permissions"
  >
    <Emojione
      text=":information_source: :point_up: Please accept Notifications to get notified when the next user is due!"
    />
  </div>
</div>
`;
